code/API_definitions/geofencing.yaml

```
@@ -0,0 +1,785 @@
+      description: Detail for event channel
+      properties:
+        notificationUrl:
+          $ref: '#/components/schemas/NotificationUrl'
+        notificationAuthToken:
+          $ref: '#/components/schemas/NotificationAuthToken'
+      required:
+        - notificationUrl
+
+    SubscriptionId:
+      type: string
+      format: uuid
+      description: Identifier of the event subscription - This attribute must not be present in the POST request as it is provided by API server
+
+    SubscriptionDetail:
+      type: object
+      description: Subscription detail
+      properties:
+        device:
+          $ref: '#/components/schemas/Device'
+        area:
+          $ref: '#/components/schemas/Area'
+        type:
+          $ref: "#/components/schemas/Type"
+      required:
+        - device
+        - area
+        - type
+
+    Type:
+      description: |
+        area-entered - Event triggered when the device enters the given area
+        area-left - Event triggered when the device leaves the given area
+        subscription-ends - Event triggered when the subscription ends
+      type: string
+      enum:
+        - org.camaraproject.geofencing.v0.area-entered
+        - org.camaraproject.geofencing.v0.area-left
+        - org.camaraproject.geofencing.v0.subscription-ends
+
+    NotificationUrl:
+      type: string
+      example: "https://application-server.com"
+      description: https callback address where the event notification must be POST-ed
+
+    NotificationAuthToken:
+      type: string
+      example: "c8974e592c2fa383d4a3960714"
+      description: "OAuth2 token to be used by the callback API endpoint. It MUST be indicated within HTTP Authorization header e.g. Authorization: Bearer $notificationAuthToken"
+
+    CreateSubscription:
+      description: The request for creating a Location Subscription
+      type: object
+      properties:
+        webhook:
+          $ref: '#/components/schemas/Webhook'
+        subscriptionDetail:
+          $ref: '#/components/schemas/SubscriptionDetail'
+        subscriptionExpireTime:
+          type: string
+          format: date-time
+          example: 2023-01-17T13:18:23.682Z
+          description: The time when the location-tracking has to be terminated (e.g. in 14 days from now on). Provided in date-time format.
+      required:
+        - webhook
+        - subscriptionDetail
+
+    SubscriptionInfo:
+      description: Represents a area-location event subscription.
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/CreateSubscription'
+        - type: object
+          properties:
+            eventSubscriptionId:
+              $ref: '#/components/schemas/SubscriptionId'
+            startsAt:
+              type: string
+              format: date-time
+              example: 2023-01-17T13:18:23.682Z
+              description: The time when the location-tracking started (e.g. in 7 days from now on). Provided in date-time format.
```

```suggestion
              description: Date when the event subscription will begin/began. Provided in date-time format.
```

I wouldn't mention location-tracking here as it may be misleading as we are not tracking devices here.